crates/testing-framework/src/transform/mod.rs

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2022 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Plan:
+ * 1. [x] line numbers and pos
+ * 2. [x] contexts for error reporting
+ * 3. [x] error report
+ * 4. [x] annotation parsing
+*/
+
+mod parser;
+pub(crate) mod walker;
+
+use crate::asserts::AssertionChain;
+
+type Triplet = (Sexp, Sexp, Sexp);
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct Call {
+    triplet: Box<Triplet>,
+    args: Vec<Sexp>,
+    var: Option<Box<Sexp>>,
+    annotation: Option<AssertionChain>,
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum Sexp {
```

Comment:
I want that `S` to be elaborated

Reply:
It's classical term for Lips-like code handled as data.
https://en.wikipedia.org/wiki/S-expression